### PR TITLE
add the interfaces

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: cpi-side/configuration/models/list.model.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { ListSearch, ListSmartSearch  } from "./search.model"
+
+
+
+export interface List{
+    Key: string,
+    Name: string,
+    Description?: string,
+    Resource: string,
+    Editor?: string,
```
Comment (roiaharon-pepperi) [inline cpi-side/configuration/models/list.model.ts:12]: didn't we said we're removing the editor? and the profile?

File: cpi-side/configuration/models/menu.model.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    Blocks: ListMenuBlock[];
+}
+
+
+export interface ListMenuBlock{
+    Key: string;
+    Title: string;
+    ButtonStyleType?: StyleType //if exist then this is a button
+    IsChangedURL: string //route to cpi side function that takes prev state and curr state and array of the menu blocks,  and determine if this menu block needs to be re rendered
+    AddonUUID: string //the addon uuid who supply the rendering function
```
Comment (roiaharon-pepperi) [inline cpi-side/configuration/models/menu.model.ts:11]: where is the property for the draw function? don't you need it here?

File: cpi-side/configuration/models/menu.model.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+
+type StyleType = 'weak' | 'weak-invert' | 'regular' | 'strong';
```
Comment (roiaharon-pepperi) [inline cpi-side/configuration/models/menu.model.ts:15]: there too should be PascalCase

File: cpi-side/configuration/models/view.model.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+interface View {
+    Key: string;
+    Type: "Grid" | "Cards";
```
Comment (roiaharon-pepperi) [inline cpi-side/configuration/models/view.model.ts:3]: you should add ViewType type, in the future there will be more types

File: cpi-side/configuration/models/view.model.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+interface ViewBlock{
+    Title: string;
+    Group: string //used to group related blocks
+    Configuration: any;
+    AddonUUID: string //the addon that exposing the draw function
+    DrawDataURL: string //the relative url that define in this relation to draw that block
+}
+
+interface GridView extends View{
```
Comment (roiaharon-pepperi) [inline cpi-side/configuration/models/view.model.ts:16]: why do we need these extensions? they are not extending anything

File: cpi-side/events/models/list-data.model.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export interface ListData{
+    Items: ItemCell[]
+}
+
+ // represent one row (i.e. one item)
+ // the matching between block and property is done by the "key" 
+ export interface ItemCell {
```
Comment (roiaharon-pepperi) [inline cpi-side/events/models/list-data.model.ts:7]: change ItemCell to RowData

File: cpi-side/events/models/list-state.model.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { JSONFilter } from "@pepperi-addons/pepperi-filters";
+
+export interface ListState {
+    ListKey: string,
+    ViewKey?: string, // default use the first
+    SearchString?: string, // default no search 
+    SmartSearchQuery?: JSONFilter
+    Sorting?: Sorting, // default CreationDateTime, Ascending=false
+    PageSize?: number //default is 50!
+    PageIndex?: number // default = 1 (first page) - does not make sense to have from to index with page index
```
Comment (roiaharon-pepperi) [inline cpi-side/events/models/list-state.model.ts:10]: refine the comment, it's no longer valid

File: cpi-side/events/models/list-state.model.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { JSONFilter } from "@pepperi-addons/pepperi-filters";
+
+export interface ListState {
+    ListKey: string,
+    ViewKey?: string, // default use the first
+    SearchString?: string, // default no search 
+    SmartSearchQuery?: JSONFilter
+    Sorting?: Sorting, // default CreationDateTime, Ascending=false
+    PageSize?: number //default is 50!
```
Comment (roiaharon-pepperi) [inline cpi-side/events/models/list-state.model.ts:9]: I think the default should be 100, 50 is too little

File: cpi-side/events/models/list-state.model.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    PageIndex?: number // default = 1 (first page) - does not make sense to have from to index with page index
+    ItemSelection?: ItemSelection // default empty 
+}
+
+interface ItemSelection{
+    SelectAll: Boolean // default false 
+    Items: string[] // key list - limit to 500 keys
```
Comment (roiaharon-pepperi) [inline cpi-side/events/models/list-state.model.ts:16]: need to refine the comment. if the select all is true, than the list on items is 'excluded' and not 'included'

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]
replace the inheritance. move Configuration & events inside model folder and not the other way around

File: cpi-side/events/models/list-state.model.ts (hunk 8 above)
Comment (michaelatias) [reply cpi-side/events/models/list-state.model.ts]: thats the default on the generic list

Comment (roiaharon-pepperi) [reply cpi-side/events/models/list-state.model.ts]: the default for the generic list is 100

**Review by roiaharon-pepperi — APPROVED** [review]

**Pull request merged**